### PR TITLE
Add root certs for nats streaming (stan) client

### PR DESCRIPTION
**Pull request by rayjanoka**
## Description

Add Root CAs from minio flag `--certs-dir` for nats streaming (stan) client.

## Motivation and Context

Minio is able to notify nats by supplying the self-signed root CA using:

Minio settings
```shell
MINIO_NOTIFY_NATS_TLS=on
MINIO_NOTIFY_NATS_TLS_SKIP_VERIFY=off
```

Helm settings
```shell
--set trustedCertsSecret=minio-trusted-certs
```

But as soon as I turn on nats streaming in minio configuration like this:
```
MINIO_NOTIFY_NATS_STREAMING=on
MINIO_NOTIFY_NATS_STREAMING_CLUSTER_ID=test
``` 

It fails with `Error: x509: certificate signed by unknown authority (x509.UnknownAuthorityError)`

I found that the `stan` client wasn't being supplied the `RootCAs` at all so I added it in the same way the regular `nats` client is configured in the lines above. Here are those lines from the `nats` client implementation: https://github.com/minio/minio/blob/master/internal/event/target/nats.go#L168-L172

## How to test this PR?

(this is a long one)

Setup nats with self-signed cert

https://github.com/nats-io/k8s/tree/main/helm/charts/nats

https://docs.nats.io/running-a-nats-service/configuration/securing_nats/tls#self-signed-certificates-for-testing

```yaml
auth:
  enabled: false
cluster:
  enabled: true
  replicas: 3
  tls:
    ca: ca.crt
    cert: tls.crt
    key: tls.key
    secret:
      name: nats-server-tls
fileStorage:
  storageClassName: nats-nats
k8sClusterDomain: cluster.local
nats:
  image: nats:2.8.4-alpine
  logging:
    debug: true
    trace: true
  resources:
    requests:
      cpu: "0.25"
      memory: 500Mi
  tls:
    ca: ca.crt
    cert: tls.crt
    key: tls.key
    secret:
      name: nats-server-tls
    verify: false
```

Setup nats-streaming with self-signed cert

https://github.com/nats-io/k8s/tree/main/helm/charts/stan

```yaml
nats:
  tls:
    ca: ca.crt
    cert: tls.crt
    key: tls.key
    secret:
      name: nats-server-tls
stan:
  clusterID: test
  logging:
    adsf: false
    debug: true
    trace: true
  nats:
    url: nats://nats.nats.svc.cluster.local:4222
  replicas: 1
  tls:
    enabled: true
    secretName: nats-streaming-server-tls
    settings:
      insecure: true
```

Add `trustedCertsSecret` with CA file (from nats) in a secret via Helm 

https://github.com/minio/minio/tree/master/helm/minio#installing-certificates-from-third-party-cas

```shell
helm upgrade --install \
  --namespace minio \
  --set mode=standalone \
  --set replicas=1 \
  --set persistence.size=5Gi \
  --set resources.requests.memory=1Gi \
  --set trustedCertsSecret=minio-trusted-certs \
  --set environment.MINIO_NOTIFY_NATS_ENABLE=on \
  --set environment.MINIO_NOTIFY_NATS_TLS=on \
  --set environment.MINIO_NOTIFY_NATS_TLS_SKIP_VERIFY=off \
  --set environment.MINIO_NOTIFY_NATS_ADDRESS="nats.nats.svc.cluster.local:4222" \
  --set environment.MINIO_NOTIFY_NATS_SUBJECT=minioevents \
  --set environment.MINIO_NOTIFY_NATS_STREAMING=on \
  --set environment.MINIO_NOTIFY_NATS_STREAMING_CLUSTER_ID=test \
  minio \
  minio/minio
```

## Types of changes
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation updated
- [ ] Unit tests added/updated

### CODE REVIEW

File: internal/event/target/nats.go [hunk 1]
```diff
@@ -187,6 +187,11 @@ func (n NATSArgs) connectStan() (stan.Conn, error) {
 	}
 
 	connOpts := []stan.Option{stan.NatsURL(addressURL)}
+
+	if n.TLS && !n.TLSSkipVerify {
+		connOpts = append(connOpts, stan.NatsOptions(nats.Secure(&tls.Config{RootCAs: n.RootCAs})))
+	}
```
Comment (harshavardhana) [inline internal/event/target/nats.go:193]: stan is deprecated by NATS folks, please move to JetStream @rayjanoka

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: internal/event/target/nats.go (hunk 1 above)
Comment (rayjanoka) [reply internal/event/target/nats.go:193]: I would like to but I didn't think minio supported it.

in this issue you seem to indicate it wasn't supported: https://github.com/minio/minio/issues/14942

@harshavardhana

Comment (harshavardhana) [reply internal/event/target/nats.go:193]: It is supported in master @rayjanoka - we deprecated stan since today's release. This PR is not needed.

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :x: [more...](#e54a5b1d-1b97-49dd-95d7-69b7a4c34272) |
<a name="e54a5b1d-1b97-49dd-95d7-69b7a4c34272"></a>
### 15241-230b9a9/mint-gateway-s3.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s8:32459
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 7 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 43 seconds
(4/15) Running aws-sdk-ruby tests ... done in 6 seconds
(5/15) Running awscli tests ... done in 51 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 24 seconds
FAILED in 7 minutes and 1 seconds
Unhandled exception. System.AggregateException: One or more errors occurred. (A task was canceled.)
 ---> System.Threading.Tasks.TaskCanceledException: A task was canceled.
   at System.Threading.Tasks.Task.GetExceptions(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at System.Threading.Tasks.Task.Wait()
   at Minio.Functional.Tests.Program.Main(String[] args) in /mint/run/core/minio-dotnet/temp/Minio.Functional.Tests/Program.cs:line 180
--- End of stack trace from previous location ---

   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at System.Threading.Tasks.Task.Wait()
   at Minio.Functional.Tests.Program.Main(String[] args) in /mint/run/core/minio-dotnet/temp/Minio.Functional.Tests/Program.cs:line 180
(8/15) Running minio-go tests ... done in 41 seconds
(9/15) Running minio-java tests ... done in 32 seconds
(10/15) Running minio-js tests ... done in 42 seconds
(11/15) Running minio-py tests ... done in 55 seconds
(12/15) Running s3cmd tests ... done in 1 minutes and 4 seconds
(13/15) Running s3select tests ... done in 3 seconds
(14/15) Running versioning tests ... done in 0 seconds

Executed 14 out of 15 tests successfully.
```
Deleting image on docker hub
{"detail": "Not found"}Deleting image locally

File: internal/event/target/nats.go (hunk 1 above)
Comment (rayjanoka) [reply internal/event/target/nats.go:193]: great news! thanks @harshavardhana!

**Comment by rayjanoka** [comment]
Move to Jetstream instead! 

https://github.com/minio/minio/releases/tag/RELEASE.2022-07-06T20-29-49Z

https://github.com/minio/minio/pull/15201

**Pull request closed without merging**